cuphead/src/lib.rs

```diff
@@ -356,3 +383,73 @@ async fn tick<'a>(
 
     Ok(())
 }
+
+pub fn monitor_star_skip(
+    memory: &Memory,
+    measured_state: &mut MeasuredState,
+) -> Result<(), Box<dyn Error>> {
+    let difficultyTickerStartedCounting = memory.difficulty_ticker_started_counting.current()?;
+    let difficultyTickerFinishedCounting = memory.difficulty_ticker_finished_counting.current()?;
+    if !difficultyTickerStartedCounting {
+        measured_state.difficulty_ticker_start_time = None;
+        measured_state.difficulty_ticker_end_time = None;
+        return Ok(());
+    }
+
+    if measured_state.difficulty_ticker_start_time.is_none() {
+        measured_state.difficulty_ticker_start_time = Some(Instant::now());
+    }
+
+    if !difficultyTickerFinishedCounting || measured_state.difficulty_ticker_end_time.is_some() {
+        return Ok(());
+    }
+
+    measured_state.difficulty_ticker_end_time = Some(Instant::now());
+
+    let start = measured_state
+        .difficulty_ticker_start_time
+        .expect("start time must exist");
+
+    let end = measured_state
+        .difficulty_ticker_end_time
+        .expect("end time must exist");
+
+    let diff: Duration = end.duration_since(start);
+    let ms = diff.as_millis() as u64;
+
+    match memory.level_difficulty.current()? {
+        Mode::Easy => {
+            if ms < 100 {
```

Comment:
You should use the Duration concept here instead of raw integers


```suggestion
    let diff: Duration = end.duration_since(start);

    match memory.level_difficulty.current()? {
        Mode::Easy => {
            if diff < Duration::from_millis(100) {
```

+ on top of that ^ you should extract out the 100ms/600ms/1100ms out to constants with meaningful names and refer to them here